contracts/consensus/Consensus.sol

```diff
@@ -570,10 +584,16 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
 
         // Join in core contract.
         (uint256 validatorCount, uint256 minValidatorCount) =
-            CoreI(_core).joinDuringCreation(msg.sender);
+            CoreI(core).joinDuringCreation(msg.sender);
 
         if (validatorCount >= minValidatorCount) {
-            coreLifetimes[_core] = CoreLifetime.genesis;
+            coreLifetimes[core] = CoreLifetime.genesis;
+            ConsensusGatewayI consensusGateway = consensusGateways[_metachainId];
```

Comment:
it should check consensusGateway exists for metachain Id?

Reply:
Done!